x-pack/plugins/security_solution/scripts/run_playwright/playwright.ts

```diff
@@ -0,0 +1,381 @@
+                ...readProviderSpec('Service', config.get('services')),
+              ]);
+
+              const options = {
+                installDir: process.env.KIBANA_INSTALL_DIR,
+                ci: process.env.CI,
+              };
+              let fleetServer: StartedFleetServer | undefined;
+              let shutdownEs;
+
+              try {
+                shutdownEs = await pRetry(
+                  async () =>
+                    runElasticsearch({
+                      config,
+                      log,
+                      name: `ftr-${esPort}`,
+                      esFrom: config.get('esTestCluster')?.from || 'snapshot',
+                      onEarlyExit,
+                    }),
+                  { retries: 2, forever: false }
+                );
+                await runKibanaServer({
+                  procs,
+                  config,
+                  installDir: options?.installDir,
+                  extraKbnOpts:
+                    options?.installDir || options?.ci || !isOpen
+                      ? []
+                      : ['--dev', '--no-dev-credentials'],
+                  onEarlyExit,
+                  inspect: argv.inspect,
+                });
+
+                if (playwrightConfigFile.env?.WITH_FLEET_SERVER) {
+                  log.info(`Setting up fleet-server for this Cypress config`);
+
+                  const kbnClient = createKbnClient({
+                    url: baseUrl,
+                    username: config.get('servers.kibana.username'),
+                    password: config.get('servers.kibana.password'),
+                    log,
+                  });
+
+                  fleetServer = await pRetry(
+                    async () =>
+                      startFleetServer({
+                        kbnClient,
+                        logger: log,
+                        port:
+                          fleetServerPort ?? config.has('servers.fleetserver.port')
+                            ? (config.get('servers.fleetserver.port') as number)
+                            : undefined,
+                        // `force` is needed to ensure that any currently running fleet server (perhaps left
+                        // over from an interrupted run) is killed and a new one restarted
+                        force: true,
+                      }),
+                    { retries: 2, forever: false }
+                  );
+                }
+
+                await providers.loadAll();
+                const functionalTestRunner = new FunctionalTestRunner(
+                  log,
+                  config,
+                  EsVersion.getDefault()
+                );
+                const ftrEnv = await pRetry(() => functionalTestRunner.run(abortCtrl.signal), {
+                  retries: 1,
+                });
+                log.debug(
+                  `Env. variables returned by [functionalTestRunner.run()]:\n`,
+                  JSON.stringify(ftrEnv, null, 2)
```

Comment:
We probably also don't want to log this.

Reply:
@jbudz done!